db/mongo_test.go

```diff
@@ -0,0 +1,302 @@
+package db
+
+import (
+	"log"
+	"net/url"
+	"os"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+var (
+	trialCreateTime = time.Date(2014, time.November, 12, 11, 45, 26, 371000000, time.UTC)
+	trialTimeZone   = "America/Los_Angeles"
+)
+
+var cron1 = CronJob{
+	ID:       "",
+	IsActive: true,
+	Function: "test1",
+	Workload: "",
+	CronTime: "* * * * * *",
+	TimeZone: trialTimeZone,
+	Created:  trialCreateTime,
+}
+
+var cron2 = CronJob{
+	ID:       "",
+	IsActive: true,
+	Function: "test1",
+	Workload: "",
+	CronTime: "1 2 3 4 5 6",
+	TimeZone: trialTimeZone,
+	Created:  trialCreateTime,
+}
+
+var cron3 = CronJob{
+	ID:       "",
+	IsActive: false,
+	Function: "test2",
+	Workload: "",
+	CronTime: "6 5 4 3 2 1",
+	TimeZone: trialTimeZone,
+	Created:  trialCreateTime,
+}
+
+var cron4 = CronJob{
+	ID:       "",
+	IsActive: true,
+	Function: "test3",
+	Workload: "",
+	CronTime: "6 5 4 3 2 1",
+	TimeZone: trialTimeZone,
+	Created:  trialCreateTime,
+}
+
+var cron5 = CronJob{
+	ID:       bson.NewObjectId().Hex(),
+	IsActive: true,
+	Function: "test4",
+	Workload: "",
+	CronTime: "6 5 4 3 2 1",
+	TimeZone: trialTimeZone,
+	Created:  trialCreateTime,
+}
+
+var cron6 = CronJob{
+	ID:       bson.NewObjectId().Hex(),
+	IsActive: true,
+	Function: "test4",
+	Workload: "{\"foo\":\"bar\"}",
+	CronTime: "6 5 4 3 2 1",
+	TimeZone: trialTimeZone,
+	Created:  trialCreateTime,
+}
+
+func TestParseWorkload(t *testing.T) {
+	w1 := "--task"
+	assert.Equal(t, w1, parseWorkload(w1))
+
+	w2 := "{\"foo\": \"bar\"}"
+	expected := bson.M(map[string]interface{}{"foo": "bar"})
+	assert.Equal(t, expected, parseWorkload(w2))
+}
+
+func TestToCronJob(t *testing.T) {
+	mc1 := mongoCronJob{
+		ID:       bson.ObjectIdHex(cron5.ID),
+		IsActive: cron5.IsActive,
+		Function: cron5.Function,
+		Workload: "",
+		CronTime: cron5.CronTime,
+		TimeZone: cron5.TimeZone,
+		Created:  cron5.Created,
+	}
+
+	assert.Equal(t, mc1.toCronJob(), cron5)
+
+	mc2 := mongoCronJob{
+		ID:       bson.ObjectIdHex(cron6.ID),
+		IsActive: cron6.IsActive,
+		Function: cron6.Function,
+		Workload: bson.M(map[string]interface{}{"foo": "bar"}),
+		CronTime: cron6.CronTime,
+		TimeZone: cron6.TimeZone,
+		Created:  cron6.Created,
+	}
+	assert.Equal(t, mc2.toCronJob(), cron6)
+
+}
+
+func TestToMongoCron(t *testing.T) {
```

Comment:
are these test cases covered implicitly by the get / create tests below? If so might be worth getting rid of them so there's a bit less code